utils/minio_utils.py

```
@@ -0,0 +1,163 @@
+
+        Returns:
+            bytes: File contents as bytes
+
+        Raises:
+            Exception: If download fails
+        """
+        try:
+            response = self.client.get_object(self.bucket_name, object_name)
+            return response.read()
+        except Exception as e:
+            raise Exception(f"Failed to download {object_name}: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)